lang/cs/Org.Apache.REEF.Tests/Functional/FaultTolerant/TestContextStart.cs

```diff
@@ -0,0 +1,238 @@
+                .Set(DriverConfiguration.OnTaskCompleted, GenericType<ContextStartDriver>.Class)
+                .Set(DriverConfiguration.OnContextClosed, GenericType<ContextStartDriver>.Class)
+                .Build();
+        }
+
+        private sealed class ContextStartDriver :
+             IObserver<IDriverStarted>,
+             IObserver<IAllocatedEvaluator>,
+             IObserver<IActiveContext>,
+             IObserver<ICompletedTask>,
+             IObserver<IClosedContext>
+        {
+            private readonly IEvaluatorRequestor _requestor;
+            private const string ContextId1 = "ContextID1";
+            private const string ContextId2 = "ContextID2";
+            private const string TaskId = "TaskID";
+            private bool _first = true;
+
+            [Inject]
+            private ContextStartDriver(IEvaluatorRequestor evaluatorRequestor)
+            {
+                _requestor = evaluatorRequestor;
+            }
+
+            public void OnNext(IDriverStarted value)
+            {
+                _requestor.Submit(_requestor.NewBuilder().Build());
+            }
+
+            public void OnNext(IActiveContext value)
+            {               
+                Logger.Log(Level.Info, "IActiveContext: " + value.Id);
+
+                if (_first)
+                {
+                    Assert.Equal(value.Id, ContextId1);
+                    _first = false;
+                    value.SubmitContext(
+                        ContextConfiguration.ConfigurationModule
+                            .Set(ContextConfiguration.Identifier, ContextId2)
+                            .Set(ContextConfiguration.OnContextStart, GenericType<ContextStartHandler>.Class)
+                            .Build());
+                }
+                else
+                {
+                    Assert.Equal(value.Id, ContextId2);
+                    var c = TaskConfiguration.ConfigurationModule
+                        .Set(TaskConfiguration.Identifier, TaskId)
+                        .Set(TaskConfiguration.Task, GenericType<TestTask>.Class)
+                        .Build();
+                    value.SubmitTask(c);
+                }
+            }
+
+            public void OnNext(IAllocatedEvaluator value)
+            {
+                value.SubmitContext(
+                    ContextConfiguration.ConfigurationModule
+                        .Set(ContextConfiguration.Identifier, ContextId1)
+                        .Set(ContextConfiguration.OnContextStart, GenericType<ContextStartHandler>.Class)
+                        .Build());
+            }
+
+            public void OnNext(ICompletedTask value)
+            {
+                Logger.Log(Level.Info, "Task is completed:" + value.Id);
+                Assert.Equal(value.Id, TaskId);
+                value.ActiveContext.Dispose();
+            }
+
+            public void OnNext(IClosedContext value)
```

Comment:
I see it now. However, as I reported on REEF-1269, I was wrong: We shouldn't actually get an `IClosedContext` for the root context. The java code doesn't create the event. I think that is a mistake and will file another JIRA for that.

For now, we should revert this code to assume that there is no `IClosedContext` for the root context.